examples/image_classification/ios/ImageClassifier/ViewContoller/ViewController.swift

```diff
@@ -0,0 +1,400 @@
+
+  // MARK: View Handling Methods
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    // Create image classifier helper
+    liveStreamDelegate = self
+    imageClassifierHelper = ImageClassifierHelper(model: model, maxResults: maxResults, scoreThreshold: scoreThreshold, runningModel: runingModel, delegate: liveStreamDelegate)
+
+    runningModelTabbar.selectedItem = cameraTabbarItem
+    runningModelTabbar.delegate = self
+    cameraCapture.delegate = self
+  }
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+#if !targetEnvironment(simulator)
+    if runingModel == .liveStream {
+      cameraCapture.checkCameraConfigurationAndStartSession()
+    }
+#endif
+  }
+
+#if !targetEnvironment(simulator)
+  override func viewWillDisappear(_ animated: Bool) {
+    super.viewWillDisappear(animated)
+    cameraCapture.stopSession()
+  }
+#endif
+
+  override var preferredStatusBarStyle: UIStatusBarStyle {
+    return .lightContent
+  }
+
+  // MARK: Storyboard Segue Handlers
+  override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+    super.prepare(for: segue, sender: sender)
+    if segue.identifier == "EMBED" {
+      inferenceViewController = segue.destination as? InferenceViewController
+      inferenceViewController?.maxResults = maxResults
+      inferenceViewController?.modelChose = model
+      inferenceViewController?.delegate = self
+      guard let inferenceVC = inferenceViewController else { return }
+      bottomViewHeightConstraint.constant = inferenceVC.collapsedHeight + inferenceBottomHeight
+      bottomSheetViewBottomSpace.constant = -inferenceBottomHeight + expandButtonHeight
+      view.layoutSubviews()
+    }
+  }
+
+  // MARK: IBAction
+
+  @IBAction func addPhotoButtonTouchUpInside(_ sender: Any) {
+    openImagePickerController()
+  }
+  // Resume camera session when click button resume
+  @IBAction func resumeButtonTouchUpInside(_ sender: Any) {
+    cameraCapture.resumeInterruptedSession { isSessionRunning in
+      if isSessionRunning {
+        self.resumeButton.isHidden = true
+        self.cameraUnavailableLabel.isHidden = true
+      }
+    }
+  }
+  // MARK: Private function
+  private func openImagePickerController() {
+    if UIImagePickerController.isSourceTypeAvailable(.savedPhotosAlbum){
+      let imagePicker = UIImagePickerController()
+      imagePicker.delegate = self
+      imagePicker.sourceType = .savedPhotosAlbum
+      imagePicker.mediaTypes = [UTType.image.identifier, UTType.movie.identifier]
+      imagePicker.allowsEditing = false
+      DispatchQueue.main.async {
+        self.present(imagePicker, animated: true, completion: nil)
+      }
+    }
+  }
+
+  private func removePlayerViewController() {
+    playerViewController.view.removeFromSuperview()
+    playerViewController.removeFromParent()
+  }
+
+  private func processVideo(url: URL) {
+    let player = AVPlayer(url: url)
```

Comment:
You're reading frames from a video file using AVPlayer, which requires a UIViewController. There's another way that shouldn't require UIViewController. Please migrate to this method.
https://stackoverflow.com/questions/42665271/swift-get-all-frames-from-video

Besides, please rename this method to `classifyVideoFile` and move it to ImageClassifierHelper like in Android. Make sure to pass the inferenceInterval as a parameter.
https://github.com/googlesamples/mediapipe/blob/39ce84ae02628d7891d8a6656c7e8de860d31df8/examples/image_classification/android/app/src/main/java/com/google/mediapipe/examples/imageclassification/ImageClassifierHelper.kt#L231

Reply:
Another reason for this change is that AVPlayer doesn't give you access to the orientation of the frame, but AVAsset does. It's required to correctly classify images videos recorded by your smartphone in portrait mode.